harmonic/install_ubuntu_src.md

```diff
@@ -0,0 +1,279 @@
+
+The first step is to create a developer workspace in which `vcstool` and
+`colcon` can work:
+
+```bash
+mkdir -p ~/workspace/src
+cd ~/workspace/src
+```
+
+All the sources of gazebo-harmonic are declared in a yaml file. Download
+it to the workspace:
+
+```bash
+wget https://raw.githubusercontent.com/gazebo-tooling/gazebodistro/master/collection-harmonic.yaml
+```
+
+Use `vcstool` to automatically retrieve all the Gazebo libraries sources from
+their repositories:
+
+```bash
+vcs import < collection-harmonic.yaml
+```
+
+The src subdirectory should contain all the sources ready to be built.
+
+## Install dependencies
+
+Before compiling it is necessary to install all the dependencies of the different
+packages that compose the Harmonic collection. Every platform has a different
+method to install software dependencies.
+
+Add `packages.osrfoundation.org` to the apt sources list:
+
+```bash
+sudo wget https://packages.osrfoundation.org/gazebo.gpg -O /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/gazebo-stable.list > /dev/null
+sudo apt-get update
+```
+
+The command below will install all dependencies in Ubuntu:
+
+```bash
+sudo apt -y install \
+  $(sort -u $(find . -iname 'packages-'`lsb_release -cs`'.apt' -o -iname 'packages.apt' | grep -v '/\.git/') | sed '/gz\|sdf/d' | tr '\n' ' ')
+```
+
+## Building the Gazebo Libraries
+
+Once the compiler and all the sources are in place it is time to compile them.
+Start the procedure by changing into the workspace and listing the packages
+recognized by `colcon`:
+
+```bash
+cd ~/workspace/
+colcon graph
+```
+
+`colcon graph` should list the Gazebo libraries with an
+[interdependency diagram](https://colcon.readthedocs.io/en/released/reference/verb/graph.html#example-output).
+If that is the case, then you are ready
+to build the whole set of libraries:
+
+```bash
+colcon build --merge-install
+```
+
+To speed up the build process, you could also disable tests by using
+
+```bash
+colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
+```
+
+To use debuggers activate debug symbols. Gazebo will run slower, but you'll be able to use GDB:
+
+```bash
+colcon build --cmake-args ' -DBUILD_TESTING=OFF' ' -DCMAKE_BUILD_TYPE=Debug' --merge-install
+```
+
+To build a specific package with all its dependent packages:
+
+```bash
+colcon build --merge-install --packages-up-to PACKAGE_NAME
+```
+
+To build a single package:
+
+```bash
+colcon build --packages-select PACKAGE_NAME
+```
+
+Visit [colcon documentation](https://colcon.readthedocs.io/en/released/#) to view more `colcon` build and test options.
+
+If there are no errors, all the binaries should be ready to use. You can check the [Troubleshooting] section for errors.
```

Comment:
link to Troubleshooting section
```suggestion
If there are no errors, all the binaries should be ready to use. You can check the [Troubleshooting](/docs/Harmonic/troubleshooting#ubuntu) section for errors.
```

Reply:
[5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)